pybamm/expression_tree/broadcasts.py

```diff
@@ -162,8 +164,17 @@ def check_and_set_domains(self, child: pybamm.Symbol, broadcast_domain: list[str
                 """Primary broadcast from particle size domain must be to particle
                 domain"""
             )
-        elif child.domain[0] in ["negative particle", "positive particle"]:
-            raise pybamm.DomainError("Cannot do primary broadcast from particle domain")
+        elif child.domain[0] in [
+            "negative particle",
+            "positive particle",
+            "negative core",
```

Comment:
There is no provision for core/shell in the negative electrode, but this might be worth keeping for future-proofing